IceCream/Classes/CKRecordConvertible.swift

```diff
@@ -128,6 +128,11 @@ extension CKRecordConvertible where Self: Object {
                 // If object is CreamAsset, set record with its wrapped CKAsset value
                 if objectName == CreamAsset.className(), let creamAsset = item as? CreamAsset {
                     r[prop.name] = creamAsset.asset
+                    let ext = creamAsset.filePath.pathExtension
+                    
+                    if !ext.isEmpty {
```

Comment:
And this judge is not enough to make ensure that `ext` is the file extension that you wanted.